derive_client/clients/ws_client.py

```diff
@@ -4,32 +4,194 @@
 
 import json
 import time
+import uuid
+from dataclasses import dataclass
+from decimal import Decimal
+from enum import StrEnum
 
+import msgspec
 from derive_action_signing.utils import sign_ws_login, utc_now_ms
-from websocket import WebSocketConnectionClosedException, create_connection
+from websockets import State
+from websockets.sync.client import ClientConnection, connect
 
+from derive_client.constants import DEFAULT_REFERER
+from derive_client.data.generated.models import (
+    Direction,
+    OrderResponseSchema,
+    PrivateGetOrdersResultSchema,
+    PrivateOrderParamsSchema,
+    PublicGetTickerResultSchema,
+    TradeResponseSchema,
+)
 from derive_client.data_types import InstrumentType, UnderlyingCurrency
+from derive_client.data_types.enums import OrderSide, OrderType, TimeInForce
 from derive_client.exceptions import DeriveJSONRPCException
 
 from .base_client import BaseClient
 
 
+@dataclass
+class Orderbook:
+    channel: str
+    timestamp: int
+    instrument_name: str
+    publish_id: int
+    bids: list[list[float]]
+    asks: list[list[float]]
+
+    @classmethod
+    def from_json(cls, data):
+        return cls(
+            channel=data["params"]["channel"],
+            timestamp=data["params"]["data"]["timestamp"],
+            instrument_name=data["params"]["data"]["instrument_name"],
+            publish_id=data["params"]["data"]["publish_id"],
+            bids=[[float(price), float(size)] for price, size in data["params"]["data"]["bids"]],
+            asks=[[float(price), float(size)] for price, size in data["params"]["data"]["asks"]],
+        )
+
+
+@dataclass
+class Position:
```

Comment:
this model should also exist in the generated models as [PositionResponseSchema](https://github.com/8ball030/derive_client/blob/95fffd3c9a0252fc7fe09ca22a1a6c866bac6f99/derive_client/data/generated/models.py#L552-L579)